pkg/server/commands/batch_check_command.go

```diff
@@ -144,30 +169,36 @@ func (bq *BatchCheckQuery) Execute(ctx context.Context, params *BatchCheckComman
 
 			response, _, err := checkQuery.Execute(ctx, checkParams)
 
-			resultMap.Store(check.GetCorrelationId(), &BatchCheckOutcome{
+			resultMap.Store(key, &BatchCheckOutcome{
 				CheckResponse: response,
 				Err:           err,
 			})
 
+			totalQueryCount.Add(response.GetResolutionMetadata().DatastoreQueryCount)
```

Comment:
Please check this, but i think that if `err != nil`, `response.GetResolutionMetadata()` will be nil too